src/control/server/engine/config.go

```diff
@@ -268,6 +269,25 @@ func (c *Config) HasEnvVar(name string) bool {
 	return false
 }
 
+// GetLineEnv return the value of the given environment variable to be supplied when starting an I/O
+// engine instance.
+func (c *Config) GetEnvVar(name string) (string, error) {
+	env, err := c.CmdLineEnv()
+	if err != nil {
+		return "", err
+	}
+
+	env = mergeEnvVars(cleanEnvVars(os.Environ(), c.EnvPassThrough), env)
```

Comment:
Why is this pulling in `os.Environ()`? The design intent was to require all engine env vars to be set explicitly by the control plane based on the contents of the configuration, not the user environment. In the past we have had multiple problems caused by users setting things in their environment and forgetting about it, then being very confused by unexpected behavior.

Reply:
From what I have understand the `os.environ()` is still used to define the environment variable used for starting the engine:
https://github.com/daos-stack/daos/blob/fe22dbe958ba62a1840952677cffcbf5b85215ad/src/control/server/engine/exec.go#L113
Thus, to have access to the same value as the engine, pulling os.environ() is still needed if I want to have the same value as the engine for the `DAOS_MD` environment variable.

Reply:
Ah, ok. Right, I'd forgotten about the pass-through option. This is an override of the normal procedure that requires all environment to be configured by the control plane, and instead allows specified user environment variables to "leak" into the engine environment.